File: 2 - Poseidon's Path Score/Solution.java
import java.util.Scanner;

public class Solution {

    public static int calcPathScore(String s) {
        int ans = 0;
        for (int i = 0; i < s.length() - 1; i++) {
            ans += Math.abs(s.charAt(i) - s.charAt(i + 1));
        }
        return ans;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        scanner.nextLine(); // Consume newline

        while (t-- > 0) {
            String s = scanner.nextLine();
            System.out.println(calcPathScore(s));
        }
        scanner.close();
    }
}

File: 2 - Poseidon's Path Score/solution.cpp
#include<bits/stdc++.h>
using namespace std;

int calcPathScore(string s) {
    int ans = 0;

    for(int i=0; i<s.length()-1; i++)
        ans += abs(s[i] - s[i+1]);

    return ans;
}

int main() {

    int t;
    cin>>t;

    while(t--) {
        string s;
        cin>>s;
        cout<<calcPathScore(s)<<endl;
    }

    return 0;
}
